File: 0872-leaf-similar-trees/0872-leaf-similar-trees.py
class Solution:
    def leafSimilar(self, root1: Optional[TreeNode], root2: Optional[TreeNode]) -> bool:
        def get_sequence(node, lst):
            if not node.left and not node.right:
                lst.append(node.val)
            if node.left:
                get_sequence(node.left, lst)
            if node.right:
                get_sequence(node.right, lst)


        seq1, seq2 = list(), list()
        get_sequence(root1, seq1)
        get_sequence(root2, seq2)

        return seq1 == seq2

File: 0872-leaf-similar-trees/0872-leaf-similar-trees.cpp
class Solution {
    vector<int> seq1;
    vector<int> seq2;

    void getSequence(TreeNode* node, vector<int>&list) {
        if (node->left == NULL && node->right == NULL) {
            list.push_back(node->val);
        }
        if (node->left != NULL) {
            getSequence(node->left, list);
        }
        if (node->right != NULL) {
            getSequence(node->right, list);
        }
    }

public:
    bool leafSimilar(TreeNode* root1, TreeNode* root2) {
        getSequence(root1, seq1);
        getSequence(root2, seq2);

        return seq1 == seq2;
    }
};